release.py

```diff
@@ -0,0 +1,224 @@
+        )
+        git.Repo.clone_from(self.repo_url, self.repo_dir, branch=branch)
+        (self.repo_dir / ".git" / "config").write_text(self.git_config)
+
+    def _check_change_lock_file(self) -> None:
+        """Check if the current version was added to the change lock file."""
+        logger.debug("Checking for change log file.")
+        if not self._change_lock_file.is_file():
+            raise Exit(
+                "Could not find change log file. "
+                f"Create one first and push it to the {self.branch} branch."
+            )
+        if f"v{self.version}" not in self._change_lock_file.read_text("utf-8"):
+            raise Exit(
+                "You need to add the version v{} to the {} change log file "
+                "and push the update to the {} branch".format(
+                    self.version,
+                    self._change_lock_file.relative_to(self.repo_dir),
+                    self.branch,
+                )
+            )
+
+    @cached_property
+    def _change_lock_file(self) -> Path:
+        """Find the change lock file."""
+        for prefix, suffix in product(("changelog", "whats-new"), (".rst", ".md")):
+            for search_pattern in (prefix, prefix.upper()):
+                glob_pattern = f"{search_pattern}{suffix}"
+                logger.debug("Searching for %s", glob_pattern)
+                for file in self.repo_dir.rglob(glob_pattern):
+                    return file
+        return Path(tempfile.mktemp())
+
+    def tag_new_version(self) -> None:
+        """Tag a new git version."""
+        self._clone_repo_from_franch(self.branch)
+        cloned_repo = git.Repo(self.repo_dir)
+        if self.version <= self.git_tag:
+            raise Exit(
+                "Tag version: {} is the same as current version {}"
+                ", you need to bump the verion number first and "
+                "push the changes to the {} branch".format(
+                    self.version,
+                    self.git_tag,
+                    self.branch,
+                )
+            )
+        self._check_change_lock_file()
+        logger.info("Creating tag for version v%s", self.version)
+        head = cloned_repo.head.reference
+        message = f"Create a release for v{self.version}"
+        try:
+            cloned_repo.create_tag(f"v{self.version}", ref=head, message=message)
+            cloned_repo.git.push("--tags")
+        except git.GitCommandError as error:
+            raise Exit("Could not create tag: {}".format(error))
+        logger.info("Tags created.")
+
+    @classmethod
+    def cli(cls, temp_dir: str) -> "Release":
+        """Command line interface."""
+
+        parser = argparse.ArgumentParser(
+            description="Prepare the release of a package."
+        )
+        subparser = parser.add_subparsers(help="Available commands:")
+        tag_parser = subparser.add_parser("tag", help="Create a new tag")
+        deploy_parser = subparser.add_parser(
```

Comment:
Yes, this is something, I am currently about to implement here: https://github.com/FREVA-CLINT/freva-deployment/blob/versions/release.py . I am thinking of downloading this file from the deployment repo instead of defining it here. We only need it when tagging stuff.